### PR TITLE
issue #1501: reconstruct to load the default value for member variables

### PR DESCRIPTION
**Pull request by zhilhuan**
fix for #1501

### CODE REVIEW

**Comment by zwoop** [comment]
[approve ci]

**Comment by zwoop** [comment]
This sounds like a potential candidate for 7.1.0 as well.

**Comment by atsci** [comment]
AU check *successful*! https://ci.trafficserver.apache.org/job/autest-github/60/

**Comment by atsci** [comment]
clang format *failed*! https://ci.trafficserver.apache.org/job/clang-format-github/63/

**Comment by atsci** [comment]
RAT check *successful*! https://ci.trafficserver.apache.org/job/RAT-github/76/

File: proxy/ControlMatcher.cc [hunk 1]
```diff
@@ -220,6 +220,7 @@ HostMatcher<Data, MatchResult>::NewEntry(matcher_line *line_info)
   if (error.failed()) {
     // There was a problem so undo the effects this function
     memset(cur_d, 0, sizeof(Data));
+    new(cur_d) Data(); //reconstruct
```
Comment (zwoop) [inline proxy/ControlMatcher.cc]: This makes sense to me, but do we still need the memset()? It's obviously harmless if not necessary, mostly curious.

**Comment by zwoop** [comment]
Please rebase with current master, and also run "make clang-format" on this.

**Comment by atsci** [comment]
Linux build *successful*! https://ci.trafficserver.apache.org/job/linux-github/1652/

**Comment by atsci** [comment]
FreeBSD11 build *successful*! https://ci.trafficserver.apache.org/job/freebsd-github/1758/

**Comment by atsci** [comment]
Intel CC build *successful*! https://ci.trafficserver.apache.org/job/icc-github/189/

**Comment by atsci** [comment]
clang-analyzer build *successful*! https://ci.trafficserver.apache.org/job/clang-analyzer-github/321/

File: proxy/ControlMatcher.cc (hunk 1 above)
Comment (zhilhuan) [inline proxy/ControlMatcher.cc]: yes, I think we do not need memset() any more. will update the commit.

**Comment by zhilhuan** [comment]
@zwoop rebased to current master and run "clang-format -i" on this file. Would you please to run the CI again?

**Comment by maskit** [comment]
[approve ci]

**Comment by atsci** [comment]
RAT check *successful*! https://ci.trafficserver.apache.org/job/RAT-github/85/

**Comment by atsci** [comment]
clang format *successful*! https://ci.trafficserver.apache.org/job/clang-format-github/72/

**Comment by atsci** [comment]
AU check *successful*! https://ci.trafficserver.apache.org/job/autest-github/69/

**Comment by atsci** [comment]
Intel CC build *failed*! https://ci.trafficserver.apache.org/job/icc-github/198/

**Comment by atsci** [comment]
Linux build *failed*! https://ci.trafficserver.apache.org/job/linux-github/1661/

**Comment by atsci** [comment]
FreeBSD11 build *successful*! https://ci.trafficserver.apache.org/job/freebsd-github/1767/

**Comment by atsci** [comment]
clang-analyzer build *successful*! https://ci.trafficserver.apache.org/job/clang-analyzer-github/330/

**Comment by zhilhuan** [comment]
Hi, sorry that I am not familiar with how this CI is running. But I've tried to "make" and "make test" in my local linux server (CentOS 7.2), there is no failure. So I guess the "Linux build failed!" error should not due to the patch I submitted.

And I also checked the history of "job/linux-github", there is some similar failures in build 1657 and 1655. So I guess the failure is caused either by code in recent master or the build environment is not stable.

Would anyone give me some suggestion what I should process then? Rebase to the latest master? Or just run the CI again?

**Comment by maskit** [comment]
@zhilhuan The build issue is already fixed on master but the commit (c251f2ba72d0520d25f8099af78a540f1586f55b) is not in your branch, I think. Could you rebase again? Then all builds should succeed.

**Comment by zhilhuan** [comment]
@maskit Rebased and done a forced push. Would you please restart the CI? Thanks!

**Comment by maskit** [comment]
[approve ci]

**Comment by atsci** [comment]
RAT check *successful*! https://ci.trafficserver.apache.org/job/RAT-github/92/

**Comment by atsci** [comment]
clang format *successful*! https://ci.trafficserver.apache.org/job/clang-format-github/79/

**Comment by atsci** [comment]
AU check *successful*! https://ci.trafficserver.apache.org/job/autest-github/76/

**Comment by atsci** [comment]
FreeBSD11 build *successful*! https://ci.trafficserver.apache.org/job/freebsd-github/1774/

**Comment by atsci** [comment]
Linux build *successful*! https://ci.trafficserver.apache.org/job/linux-github/1667/

**Comment by atsci** [comment]
Intel CC build *successful*! https://ci.trafficserver.apache.org/job/icc-github/205/

**Comment by atsci** [comment]
clang-analyzer build *successful*! https://ci.trafficserver.apache.org/job/clang-analyzer-github/337/

**Pull request merged**